packages/glimmer-reference/lib/iterable.ts

```diff
@@ -152,9 +160,9 @@ export class ReferenceIterator {
 }
 
 export interface IteratorSynchronizerDelegate {
-  retain(key: InternedString, item: PathReference<any>);
-  insert(key: InternedString, item: PathReference<any>, before: InternedString);
-  move(key: InternedString, item: PathReference<any>, before: InternedString);
+  retain(key: InternedString, item: PathReference<any>, memo: PathReference<any>);
```

Comment:
Good question. In general the answer is probably "this code predates the introduction of `Opaque`", so it might be time to fix it!